src/trace/implementations/merge_batcher.rs

```diff
@@ -1,31 +1,35 @@
 //! A general purpose `Batcher` implementation based on radix sort.
 
 use std::collections::VecDeque;
+use std::marker::PhantomData;
 
 use timely::communication::message::RefOrMut;
 use timely::logging::WorkerIdentifier;
 use timely::logging_core::Logger;
 use timely::progress::frontier::AntichainRef;
 use timely::progress::{frontier::Antichain, Timestamp};
 use timely::{Container, PartialOrder};
+use timely::container::{ContainerBuilder, PushInto};
 
-use crate::consolidation::consolidate_updates;
 use crate::difference::Semigroup;
 use crate::logging::{BatcherEvent, DifferentialEvent};
 use crate::trace::{Batcher, Builder};
 use crate::Data;
 
 /// Creates batches from unordered tuples.
-pub struct MergeBatcher<M, T>
+pub struct MergeBatcher<Input, C, M, T>
 where
-    M: Merger,
+    C: ContainerBuilder<Container=M::Chunk> + Default,
```

Comment:
Is `Input` really necessary as part of the struct, or can the `PushInto<Input>` constraint be added somewhere? Alternately, could it be added here to signpost what it is and why it exists (or just doc it).

Reply:
In principle, it can be removed. Then we'd only carry a `PushInto` bound for `C` with us and everything should work. But .. for flatcontainers, Rust has a hard/impossible time to figure out what the scope's timestamp type would be and I couldn't figure out how to convince it to carry that information. Maybe it's a matter of trying a bit harder, but I feel we can do that outside of this change?